textacy/fileio/read.py

```diff
@@ -170,15 +271,29 @@ def get_filenames_in_dir(dirname, file_type=None, subdirs=False):
     if not os.path.exists(dirname):
         raise IOError('directory {} does not exist'.format(dirname))
 
-    for dirpath, dirnames, filenames in os.walk(dirname):
-        if dirpath.startswith('.'):
-            continue
-        for filename in filenames:
-            if filename.startswith('.'):
+    def is_good_file(filename, filepath):
+        if ignore_invisible and filename.startswith('.'):
+            return False
+        if match_substr and match_substr not in filename:
+            return False
+        if ignore_substr and ignore_substr in filename:
+            return False
+        if extension and not os.path.splitext(filename)[-1] == extension:
+            return False
+        if not os.path.isfile(os.path.join(filepath, filename)):
+            return False
+        return True
+
+    if recursive is True:
+        for dirpath, _, filenames in os.walk(dirname):
+            if ignore_invisible and dirpath.startswith('.'):
```

Comment:
This line checks for invisible _directories_, and skips everything under them if ignore_invisible is True. `is_good_file` checks files; there are often invisible files in visible directories. Is that unintuitive?